Portal 1.19.0 RNs.mdx (#2704)
TykTechnologies/tyk-docs: developer-support/release-notes/portal.mdx

---

## 1.19 Release Notes

### 1.19.0 Release Notes 

#### Release Date xx August 2026

#### Release Highlights

> Following the temporary exclusion of MCP Proxies flagged in earlier releases, Tyk Enterprise Developer Portal now provides first-class support for Tyk MCP Proxy APIs. MCP APIs and their products now sync into the Portal catalogue automatically, admins can add documentation for them from Portal Admin, and developers can find them in the live catalogue alongside REST, GraphQL and other supported API types, complete with a dedicated MCP badge.

Reviewer: ```suggestion
Tyk Developer Portal now provides first-class support for Tyk MCP Proxies. MCP Proxies now sync to Portal automatically, MCP Products can be created, complete with documentation and published in Catalogues. Developers can find and request access to MCP Products in the live catalogue alongside REST, GraphQL and other supported API types, complete with a dedicated MCP badge.
```

> Developers can now explore and call an MCP server directly from its documentation page in Portal, without separate tooling. This is powered by a new embedded, MCPP-compatible reverse proxy that authenticates using the developer's Portal session, applies per-user rate limiting, and blocks server-side request forgery (SSRF) by default, together with the official MCP Inspector, now vendored and embedded on product documentation pages.

Reviewer: ```suggestion
Developers can now explore and test an MCP Proxy directly from its documentation page in Portal, without separate tooling. This is powered by a new embedded, MCPP-compatible reverse proxy that authenticates using the developer's Portal session, applies per-user rate limiting, and blocks server-side request forgery (SSRF) by default, together with the official MCP Inspector, now vendored and embedded on product documentation pages.
```

Is MCPP a typo?
I would move the implementation detail into the change log - that shouldn't be in the release highlight.

> For MCP products protected with OAuth 2.0, Portal now surfaces Protected Resource Metadata (RFC 9728) - the resource identifier, authorization servers, supported scopes and discovery path - or indicates when this is discovered at runtime instead. Dynamic Client Registration also now supports public clients that authenticate with PKCE rather than a client secret.

Reviewer: ```suggestion
For MCP Products protected with OAuth 2.0, Portal now surfaces Protected Resource Metadata (RFC 9728) - the resource identifier, authorization servers, supported scopes and discovery path - or indicates when this is discovered at runtime instead. Dynamic Client Registration also now supports public clients that authenticate with PKCE rather than a client secret.
```

> Other changes in this release include updates to the Product Administration UI, with "API Products" renamed to "Products" and the "API's" tab renamed to "Services", plus consistent auth-type badge colours across the catalogue and admin.

Reviewer: ```suggestion
```

This is detail for the change log, not the release highlights.

> This release also adds support for referencing secrets and configuration values in the Portal configuration file from an external secret store, using the same unified `kv://` / `$kv{}` syntax introduced across other Tyk components this wave, and updates the Go toolchain to Go 1.26. Two bugs are also fixed: developer email addresses are now handled case-insensitively throughout Portal, and creating a product via the Portal API no longer inherits unrelated API documentation from another product.

Reviewer: ```suggestion
This release adds [Secrets Management](/tyk-configuration-reference/kv-store) to the Tyk Developer Portal configuration file for the first time, letting you reference values from external secrets stores in place of plain-text values.
```

There are no deprecations in this release.

#### Upgrade instructions
<a id="upgrade-1.19.0"></a>

<a id="upgrade-1.19.0"></a>

If you are upgrading to 1.19.0, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker Image to pull](https://hub.docker.com/r/tykio/portal/tags?page=&page_size=&ordering=&name=v1.19.0)
  - ```bash
    docker pull tykio/portal:v1.19.0
    ```
- Helm charts
  - [tyk-charts](/developer-support/release-notes/helm-chart)

#### Changelog
<a id="Changelog-v1.19.0" data-scroll-offset></a>

##### Changed

<AccordionGroup>

<Accordion title='Renamed "API Products" to "Products" and updated auth-type badge colours'>
{/* TT-17330 */}
References to "API Products" in Portal Admin are now labelled "Products", and the "API's" tab is now labelled "Services". Auth-type badges in the catalogue and admin now use a consistent colour palette across API types.
</Accordion>

<Accordion title='Updated Golang version to 1.26'>
{/* TT-17868 */}
Tyk Portal has been updated to build with Go 1.26, keeping the toolchain current with upstream Go security and performance updates.
</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>

> <Accordion title='MCP Proxy APIs and their products now sync into the Portal catalogue'>

Reviewer: ```suggestion
<Accordion title='Full support for Tyk MCP Proxies'>
```

> {/* TT-17195 */}

Reviewer: ```suggestion
{/* TT-17195, TT-17196 */}
```

> The Portal's sync process now recognises Tyk MCP Proxy APIs using the same detection Tyk Gateway uses, and syncs them along with their products and plans. This removes the temporary exclusion introduced in earlier releases.

Reviewer: ```suggestion
The Portal now synchronizes Tyk MCP Proxies from Tyk Dashboard and supports creation of Products to publish them in the Live Portal, complete with admin provided documentation
```

</Accordion>

> <Accordion title='Admins can add documentation to MCP APIs from Portal Admin'>

Reviewer: ```suggestion
```

Delete this entry - merged into the previous

{/* TT-17196 */}
A new "MCP (Model Context Protocol)" documentation type is available when adding documentation to a product, with a Markdown editor for authoring content.
</Accordion>

<Accordion title='Added an embedded reverse proxy for testing MCP servers'>
{/* TT-17197 */}
Portal now includes an embedded, MCPP-compatible reverse proxy so developers can connect to and test MCP servers directly through the Portal. It authenticates using the developer's Portal session, applies per-user rate limiting, and blocks requests to private, loopback, link-local and cloud metadata addresses by default.
</Accordion>

<Accordion title='Embedded the MCP Inspector on MCP product documentation pages'>
{/* TT-17198 */}

> The official MCP Inspector is now embedded directly on MCP product documentation pages, letting developers explore and test an MCP server's tools, resources and prompts without leaving Portal.

Reviewer: ```suggestion
The official MCP Inspector is now embedded directly on MCP Product documentation pages, letting developers explore and test an MCP server's tools, resources and prompts without leaving Portal.
```

</Accordion>

<Accordion title='MCP products now appear in the live developer catalogue'>
{/* TT-17365 */}

> MCP products are now shown in the live Portal catalogue and product detail pages, with a dedicated MCP badge and a documentation view built for MCP APIs. Products that combine MCP with other API types show a badge and documentation link for each type.

Reviewer: ```suggestion
MCP Products are now shown in the live Portal catalogue and Product detail pages, with a dedicated MCP badge and a documentation view built for MCP APIs. Products that combine MCP with other API types show a badge and documentation link for each type.
```

</Accordion>

> <Accordion title='Added Protected Resource Metadata and public OAuth client support for MCP products'>

Reviewer: Is this only for MCP Products? I thought the new OAuth 2.0 auth is for any proxy (API or MCP)

{/* TT-17591 */}
Portal now surfaces OAuth 2.0 Protected Resource Metadata (RFC 9728) for MCP products that have it enabled at the Gateway, or indicates that it's discovered at runtime for products using mirror mode. Dynamic Client Registration also supports registering public OAuth clients that authenticate using PKCE instead of a client secret.
</Accordion>

> <Accordion title='Added a unified `kv://` syntax for referencing secrets and configuration values'>

Reviewer: ```suggestion
<Accordion title='Secrets Management'>
```

{/* TT-16268, TT-17701 */}

> Tyk Portal now supports the same unified syntax for referencing secrets and configuration values held in an external store that other Tyk components use: `kv://<store-name>/<path>` to substitute a whole value, or `$kv{<store-name>:<path>}` to substitute a value inline within a string. Declare one or more named secret stores in a new `kv` section of the Portal configuration file — in `.conf`/`.json`, `.yaml`/`.yml`, or `.toml` format — then reference them from any string configuration value.

Reviewer: ```suggestion
Tyk Portal now supports Secrets Management, letting you reference values from external secrets stores: the Tyk config file, environment variables, local files (including Kubernetes-mounted secrets), Consul, Vault, AWS Secrets Manager, Azure Key Vault, and GCP Secret Manager, in place of plain-text values in Portal's configuration, using a consistent reference syntax:

- `kv://<store>/<path>[#fragment]` for whole-value references.
- `$kv{<store>:<path>[#fragment]}` for embedding a reference inside a larger string.

The `kv` configuration block can itself contain references, for example pulling a Vault token from an environment variable.

References are resolved once, at startup. Individual secrets stores can be marked as `required` and, if they cannot be reached or initialized, Portal will fail to start with a clear error rather than starting with that store silently unavailable.

See [Secrets Management](/docs/tyk-configuration-reference/kv-store) for full details.
```

> Supported store types include Hashicorp Vault, Hashicorp Consul, environment variables, local files, inline values, and cloud provider secret managers (AWS, GCP, and Azure). References are resolved once at startup. Provider configuration keys are matched case-insensitively but punctuation-exact, while store names and secret key names are matched exactly — a mismatch always fails startup with a clear error rather than silently resolving to a missing or incorrect value. A secret store marked as `required` causes startup to fail if it's unreachable.

Reviewer: ```suggestion
```

> To help verify configuration without exposing secrets, Portal also adds a `--show-configuration` CLI flag and an admin `/config` HTTP endpoint. Both display the resolved configuration with each configured secret store's own provider credentials redacted, while still showing the store's name, type, and `required` flag. Configurations with no `kv` section behave exactly as before.

Reviewer: ```suggestion
The --show-configuration CLI flag does not expose the `kv` section: once secret-store references are resolved at startup, the KV store definitions (including credentials) are cleared from the in-memory configuration, so nothing about them appears in the output.
```

This was factually incorrect!

</Accordion>

</AccordionGroup>

##### Fixed

<AccordionGroup>

> <Accordion title='Fixed case-sensitive handling of developer email addresses'>

Reviewer: ```suggestion
<Accordion title='Fixed case-sensitive handling of user email addresses'>
```

{/* TT-14811 */}

> Tyk Portal previously stored and compared developer email addresses case-sensitively, which could prevent password reset emails from being delivered and could result in duplicate accounts for the same address in a different case — for example after an admin invitation, during self-registration, or via SSO. Email address matching and lookups throughout Portal are now case-insensitive, and newly created or updated accounts store their email address in normalised (lowercase) form. This fix applies without requiring a data migration for existing accounts.

Reviewer: ```suggestion
Tyk Portal previously stored and compared user email addresses case-sensitively, which could prevent password reset emails from being delivered and could result in duplicate accounts for the same address in a different case — for example after an admin invitation, during self-registration, or via SSO.

Email address matching and lookups throughout Portal are now case-insensitive, and newly created or updated accounts store their email address in normalised (lowercase) form. This fix applies without requiring a data migration for existing accounts.
```

</Accordion>

> <Accordion title="Fixed a new product incorrectly inheriting another product's API documentation">

Reviewer: ```suggestion
<Accordion title="Fixed new Product incorrectly inheriting another Product's API documentation">
```

{/* TT-15912 */}

> When creating a new product via the Portal API, if the product referenced an API that already had documentation attached to it in a different, unrelated product, Portal would automatically link that documentation (for example an OpenAPI or GraphQL SDL document) to the new product, even though it was never submitted in the request. This didn't happen when creating a product through the Portal Admin UI. The Portal API and Portal Admin UI now behave consistently: creating or updating a product only associates documentation that is explicitly submitted for it.

Reviewer: ```suggestion
When creating a new Product via the Portal API, if the Product referenced an API that already had documentation attached to it in a different, unrelated Product, Portal would automatically link that documentation (for example an OpenAPI or GraphQL SDL document) to the new Product, even though it was never submitted in the request. This didn't happen when creating a Product through the Portal Admin UI.

The Portal API and Portal Admin UI now behave consistently: creating or updating a Product only associates documentation that is explicitly submitted for it.
```

</Accordion>

<Accordion title='Added Protected Resource Metadata and public OAuth client support for MCP products'>
{/* TT-17591 */}
Portal now surfaces OAuth 2.0 Protected Resource Metadata (RFC 9728) for MCP products that have it enabled at the Gateway, or indicates that it's discovered at runtime for products using mirror mode. Dynamic Client Registration also supports registering public OAuth clients that authenticate using PKCE instead of a client secret.
</Accordion>

<Accordion title='Secrets Management'>
{/* TT-16268, TT-17701 */}
Tyk Portal now supports Secrets Management, letting you reference values from external secrets stores: the Tyk config file, environment variables, local files (including Kubernetes-mounted secrets), Consul, Vault, AWS Secrets Manager, Azure Key Vault, and GCP Secret Manager, in place of plain-text values in Portal's configuration, using a consistent reference syntax:

- `kv://<store>/<path>[#fragment]` for whole-value references.
- `$kv{<store>:<path>[#fragment]}` for embedding a reference inside a larger string.

> The `kv` configuration block can itself contain references, for example pulling a Vault token from an environment variable.

Reviewer: ```suggestion
The secrets management configuration block (`kv`) can itself contain references, for example pulling a Vault token from an environment variable.
```
